src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/ModuleBuilder.Mono.cs

```diff
@@ -354,12 +355,14 @@ public EnumBuilder DefineEnum(string name, TypeAttributes visibility, Type under
         }
 
         [ComVisible(true)]
+        [RequiresUnreferencedCode("Types might be removed")]
```

Comment:
This seems to conflict with the suppress message on `GetTypeToken`. The ILLinker can't trim IL that is produced by Reflection.Emit, so how could a type inside a ModuleBuilder get removed?

The same question for similar annotations below.

Reply:
This will be forced by mono/linker#1271. This is an override of `Module.GetType` that is annotated as `RequiresUnreferencedCode`.

This one is already forced because the call to `GetType` within this method body is triggering a warning, but even if we were to shut the warning up, we'll get a new warning due to inconsistent annotation across virtuals once the above issue is implemented.

Reply:
This is basically a limitation of the linker - it can't do per-instance data flow analysis, it only does it per-type/method. In this case some instances of `Module.GetType` are trimmer unfriendly (`RuntimeModule.GetType` is a good example) while others are trimmer friendly in principal (like this one). Unfortunately the base abstract method `Module.GetType` must be annotation one way or another (since linker won't be able to tell which actual override is being used on any particular call site). For reflection we chose to annotate the base classes for "runtime reflection" so basically the way the `RuntimeModule` should be annotated.

Unfortunate side-effect is that any other implementation of `Module` must be annotated the same way - `ModuleBuilder` is one such example. Even worse example is going to be `MetadataLoadContext` and its implementations of reflection APIs - those are all inherently trimmer friendly, but we have no way to annotate them as such.